src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts

```diff
@@ -123,10 +108,38 @@ export class NotificationService {
           application
         );
       }
+      this.queueToShow(config);
+    }, 10);
+  }
+
+  private queueToShow(config: CdNotificationConfig) {
+    this.cancel(this.queuedTimeoutId);
+    if (!this.queued.find((c) => _.isEqual(c, config))) {
+      this.queued.push(config);
+    }
+    this.queuedTimeoutId = window.setTimeout(() => {
+      this.showQueued();
+    }, 250);
+  }
+
+  private showQueued() {
+    const byTitle = {};
```

Comment:
Why not typing this? And is `const` a valid qualifier for a mutable variable?

```ts
var byTitle: { [key:string]:CdNotificationConfig[]; } = {}
```

Reply:
`const` is fine to be used for objects.
`const` will not allow the variable name to be reassigned and will have the same scope as `let` - from the position they were created until the end of the function. `var` resembles a more global long lived scope - it will be created before anything is executed in a function and will live through out all cycles.

For example if you create a variable through `var` inside a for loop it will be created through the interpreter on the beginning of the function that uses the loop, where if you would used `let` it would only live during the loop.

But I can add the type definition that I omitted.

Reply:
Thanks for the info. Don't worry, no need to change that. I was assuming strict `const`-ness as in C++, but as there are no pointers in JS and arrays are mutable (not literals), I guess it's more or less the same as in C++...

Reply:
I refactored the method it's now easier to read ;)